File: app/src/main/java/com/kirchhoff/example/githubclient/ui/commit/CommitsContract.java
package com.kirchhoff.example.githubclient.ui.commit;

import android.support.annotation.NonNull;

import com.kirchhoff.example.githubclient.model.CommitResponse;
import com.kirchhoff.example.githubclient.ui.BasePresenter;
import com.kirchhoff.example.githubclient.ui.BaseView;

import java.util.List;

public interface CommitsContract {

    interface View extends BaseView {

        void openCommits(@NonNull List<CommitResponse> commitList);

        void showEmptyView();
    }


    interface Presenter extends BasePresenter {

        void loadCommits(@NonNull String repository);

    }
}

File: app/src/main/java/com/kirchhoff/example/githubclient/ui/commit/CommitsContract.kt
package com.kirchhoff.example.githubclient.ui.commit

import com.kirchhoff.example.githubclient.model.CommitResponse
import com.kirchhoff.example.githubclient.ui.BasePresenter
import com.kirchhoff.example.githubclient.ui.BaseView

/**
 * @author Kirchhoff-
 */
interface CommitsContract {

    interface View : BaseView {

        fun showCommits(commitList: List<CommitResponse>)

        fun showEmptyView()
    }

    interface Presenter : BasePresenter {

        fun loadCommits(repository: String)
    }
}